ts/components/helpers/withErrorModal.tsx

```diff
@@ -0,0 +1,161 @@
+import { Option } from "fp-ts/lib/Option";
+import { Button, H2, Text } from "native-base";
+import * as React from "react";
+import { Image, Modal, StyleSheet, View } from "react-native";
+import { connect } from "react-redux";
+
+import I18n from "../../i18n";
+import { Action, ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+import variables from "../../theme/variables";
+
+export type WithErrorModalInjectedProps = {
+  error: Option<string>;
+};
+
+export type WithErrorModalProps<P> = P &
+  WithErrorModalInjectedProps &
+  ReduxProps;
+
+const styles = StyleSheet.create({
+  contentWrapper: {
+    flex: 1,
+    padding: variables.contentPadding
+  },
+
+  imageAndMessageContainer: {
+    flex: 1,
+    justifyContent: "center",
+    alignItems: "center"
+  },
+
+  image: {
+    marginBottom: 20
+  },
+
+  buttonsContainer: {
+    flexDirection: "row",
+    marginTop: "auto"
+  },
+
+  buttonCancel: {
+    flex: 4,
+    backgroundColor: variables.brandDarkGray
+  },
+
+  buttonCancelText: {
+    color: variables.colorWhite
+  },
+
+  separator: {
+    width: 10
+  },
+
+  buttonRetry: {
+    flex: 8
+  }
+});
+
+/**
+ * A HOC to display a modal to notify the user an error occurred.
+ * Inside the modal cancel and retry buttons are rendered conditionally.
+ *
+ * @param WrappedComponent The react component you want to wrap
+ * @param errorSelector A redux selector that returns the error (as string) or undefined
+ * @param errorMessage The error message to show when errorSelector returns a value
```

Comment:
Quick question: the way I understand it, the selector is only used to check whether an error has occurred. Then, `errorMessage` is displayed regardless of what `errorSelector` contains. Is that correct? 

What about having `errorMessage` be a map of error_code -> error_message, and then if the selector extracts a code, `errorMessage` maps the code to the correct (& internationalized) message? (maybe with a default option in case the code found on the store is not in the keys of `errorMessage`)

Reply:
What you mean with error_code? The http response error code (like 500) or something custom written by saga after fetch requests?

Reply:
@cloudify what you think about this?

Reply:
As an example, pagoPA might fail for a bunch of different reasons. It probably makes sense to store the reason in the store (e.g. `PAYMENT_ALREADY_PROCESSED`, `INVALID_RPT_ID`), and then have `errorMessage` be a map `{ "PAYMENT_ALREADY_PROCESSED": I18n.t("..."), "INVALID_RPT_ID": I18n.t("..."), "default": I18n.t("...") }`